### ISSUE TITLE
[docs suggestion] Import order of global styles

### ISSUE THREAD

**Issue by geelen** [styled-components/styled-components#1109]
Just hit something demo-ing SC to a mate. If you do this:

```js
import { injectGlobal } from 'styled-components'
import App from './App'

injectGlobal`
  * { all: initial; }
`

ReactDOM.render(<App />, document.getElementById('root'))
```

Because this file depends upon its imports, `App.js` is _always_ evaluated before this file. Which means if it declares any styled components they will be inserted before the `injectGlobal` call here. It means something like a reset is running in the wrong order. You just need to extract another import:

```js
import './global-styles'
import App from './App'

ReactDOM.render(<App />, document.getElementById('root'))
```

```js
/* global-styles.js */
import { injectGlobal } from 'styled-components'

injectGlobal`
  * { all: initial; }
`
```

Then, order is preserved. This should probably be somewhere in the docs. Maybe it is already? Who knows, not me!

**Comment by k15a** [comment]
I think this is pretty much related to https://github.com/styled-components/styled-components/issues/1108 and we should.change the behaviour :)

**Comment by mxstbr** [comment]
Superseded by #1333, which will solve this once and for all.

**Comment by tysonmatanich** [comment]
Just wasted several hours because of this. I understand a fix is coming, however adding it to the documentation until then is the right thing to do.

**Comment by probablyup** [comment]
@tysonmatanich what are you seeing? The behavior has changed a lot since this issue. Please open a new one if you have a specific bug to report.

**Comment by tysonmatanich** [comment]
@probablyup Not a bug, just think it would be good to call out in the [documentation](https://www.styled-components.com/docs/api#injectglobal). The issue I encountered seems exactly the same as @geelen reported. The workaround of needing the global styles import before the App is imported works great. I'm using SSR if that impacts anything. Tried on both v3.2.5 and v3.2.6.

I thought `injectGlobal` just had to be called before `<App />`  was rendered but that is not the case. The styles import has to occur before the App import. 

**Works**
```
import '../styles';
import App from '../app';
```
